types/lib/model.d.ts

```diff
@@ -634,17 +634,15 @@ export interface CountOptions<TAttributes = any>
 /**
  * Options for Model.count when GROUP BY is used
  */
-export interface CountWithOptions<TAttributes = any> extends CountOptions<TAttributes> {
-  /**
-   * GROUP BY in sql
-   * Used in conjunction with `attributes`.
-   * @see Projectable
-   */
-  group: GroupOption;
-}
+export type CountWithOptions<TAttributes = any> = SetRequired<CountOptions<TAttributes>, 'group'>
 
 export interface FindAndCountOptions<TAttributes = any> extends CountOptions<TAttributes>, FindOptions<TAttributes> { }
 
+interface GroupedCountResultItem {
```

Comment:
I did not export this type. Prefer to keep it internal because the current projected attributes type is not ideal in sequelize.